internal/cli/command/cluster/feature/features/dns/activate.go

```diff
@@ -125,3 +165,249 @@ func validateActivateClusterFeatureRequest(req interface{}) error {
 
 	return validateSpec(request.Spec)
 }
+
+func buildCustomDNSFeatureRequest(banzaiCli cli.Cli, req *pipeline.ActivateClusterFeatureRequest) error {
+	domainFilters, err := askDomainFilter()
+	if err != nil {
+		return err
+	}
+
+	clusterDomain, err := askDomain()
+	if err != nil {
+		return err
+	}
+
+	provider, err := askDnsProvider()
+	if err != nil {
+		return err
+	}
+
+	providerSpec, err := askDnsProviderSpecificOptions(banzaiCli, provider)
+	if err != nil {
+		return err
+	}
+
+	req.Spec = obj{
+		"customDns": obj{
+			"enabled":       true,
+			"domainFilters": domainFilters,
+			"clusterDomain": clusterDomain,
+			"provider":      providerSpec,
+		},
+	}
+
+	return nil
+}
+
+func askDomainFilter() ([]string, error) {
+	var domainFilter string
+	if err := survey.AskOne(
+		&survey.Input{
+			Message: "Please provide a domain filter to match domains against",
+			Default: "",
+			Help:    "To add multiple domains separate with commna (,) character. Like: foo.com, bar.com",
+		},
+		&domainFilter,
+	); err != nil {
+		return nil, errors.WrapIf(err, "failure during survey")
+	}
+
+	filters := make([]string, len(domainFilter))
```

Comment:
that line doesn't do what you think it does

Reply:
sorry, you're right, fixed